### PR TITLE
Redesign location card UI

### LINKED ISSUE #684: [DepShield] (CVSS 7.4) Vulnerability due to usage of lodash.memoize:4.1.2
**Vulnerabilities**

DepShield reports that this application's usage of [lodash.memoize:4.1.2](https://ossindex.sonatype.org/component/pkg:npm/lodash.memoize@4.1.2) results in the following vulnerability(s):

- (CVSS **7.4**) [CWE-471: Modification of Assumed-Immutable Data (MAID)](https://ossindex.sonatype.org/vuln/0f23ff35-235f-404f-8118-bc1580673fd0)
- (CVSS **6.5**) [[CVE-2018-3721] lodash node module before 4.17.5 suffers from a Modification of Assumed-Immutabl...](https://ossindex.sonatype.org/vuln/12e63c9c-b3f9-42d3-8541-dca1b72cad69)

<hr />

**Occurrences**

lodash.memoize:4.1.2 is a transitive dependency introduced by the following direct dependency(s):

• **[cssnano:4.1.10](https://ossindex.sonatype.org/component/pkg:npm/cssnano@4.1.10)**
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;└─ [cssnano-preset-default:4.0.7](https://ossindex.sonatype.org/component/pkg:npm/cssnano-preset-default@4.0.7)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;└─ [postcss-merge-rules:4.0.3](https://ossindex.sonatype.org/component/pkg:npm/postcss-merge-rules@4.0.3)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;└─ [caniuse-api:3.0.0](https://ossindex.sonatype.org/component/pkg:npm/caniuse-api@3.0.0)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;└─ [lodash.memoize:4.1.2](https://ossindex.sonatype.org/component/pkg:npm/lodash.memoize@4.1.2)

• **[react-phone-input-2:2.13.7](https://ossindex.sonatype.org/component/pkg:npm/react-phone-input-2@2.13.7)**
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;└─ [lodash.memoize:4.1.2](https://ossindex.sonatype.org/component/pkg:npm/lodash.memoize@4.1.2)

<sub>This is an automated GitHub Issue created by Sonatype DepShield. Details on managing GitHub Apps, including DepShield, are available for [personal](https://help.github.com/articles/managing-your-personal-account-s-apps/) and [organization](https://help.github.com/articles/managing-your-organization-s-apps/) accounts. Please submit questions or feedback about DepShield to the [Sonatype DepShield Community](https://community.sonatype.com/c/depshield).</sub>

### LINKED ISSUE #6843: UI for Location Management Page
follow-up : #6592 

image

- [ ]  Create a badge to show location type in the location list page
- [ ]  create a mock up entire UI of the location list where location name, location description, middleware address, buttons, location type, created date and modified date

### PR DESCRIPTION
**Pull request by GokulramGHV**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 47c1b4b</samp>

This pull request enhances the location management feature of the facility module by adding a responsive grid layout, showing more location details, and updating the `LocationModel` interface to match the backend API. The changes affect the `LocationManagement.tsx` and `models.tsx` files.

## Proposed Changes

- Fixes #6843

## Screenshot
image

 

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 47c1b4b</samp>

*  Redesign the `Location` component to show more information and use a grid layout ([link](https://github.com/coronasafe/care_fe/pull/6853/files?diff=unified&w=0#diff-042c54113f687e89c2429ff1d97fe7dfcccfa3818279170a82ea1286684a53cdL54-R59), [link](https://github.com/coronasafe/care_fe/pull/6853/files?diff=unified&w=0#diff-042c54113f687e89c2429ff1d97fe7dfcccfa3818279170a82ea1286684a53cdL72-R126), [link](https://github.com/coronasafe/care_fe/pull/6853/files?diff=unified&w=0#diff-e2841f01e53119db63c01994d866100758f06ecc85d11914eecc7cdfe8a2f4dfL207-R212))
  - Add the `location_type`, `created_date`, and `modified_date` properties to the `LocationModel` interface in `models.tsx` ([link](https://github.com/coronasafe/care_fe/pull/6853/files?diff=unified&w=0#diff-e2841f01e53119db63c01994d866100758f06ecc85d11914eecc7cdfe8a2f4dfL207-R212))
  - Display the `location_type` as a badge next to the name, the `middleware_address` as a separate section, and the dates at the bottom of the `Location` component in `LocationManagement.tsx` ([link](https://github.com/coronasafe/care_fe/pull/6853/files?diff=unified&w=0#diff-042c54113f687e89c2429ff1d97fe7dfcccfa3818279170a82ea1286684a53cdL72-R126))
  - Reposition and style the `Edit` and `Manage Beds` buttons in the `Location` component ([link](https://github.com/coronasafe/care_fe/pull/6853/files?diff=unified&w=0#diff-042c54113f687e89c2429ff1d97fe7dfcccfa3818279170a82ea1286684a53cdL72-R126))
  - Use a `grid` layout with responsive columns for the `PaginatedList.Items` component and wrap it in a `div` with a `w-full @container` class in `LocationManagement.tsx` ([link](https://github.com/coronasafe/care_fe/pull/6853/files?diff=unified&w=0#diff-042c54113f687e89c2429ff1d97fe7dfcccfa3818279170a82ea1286684a53cdL54-R59))

### CODE REVIEW

File: src/Components/Facility/LocationManagement.tsx [hunk 1]
```diff
@@ -69,43 +70,60 @@ const Location = ({
...
+      Manage Beds
+    </ButtonV2>
+
+    <div className="mt-3 flex items-center justify-between gap-4 text-sm font-medium text-gray-700">
+      <p>
+        <strong>Created:</strong> {created_date?.slice(0, 10)}
+      </p>
+      <p>
+        <strong>Modified:</strong> {modified_date?.slice(0, 10)}
+      </p>
```
Comment (rithviknishad) [inline src/Components/Facility/LocationManagement.tsx]: We generally show only relative time (and absolute time in tooltip on hover) using `RecordMeta` component.

File: src/Components/Facility/models.tsx [hunk 2]
```diff
@@ -204,9 +204,12 @@ export interface LocationModel {
   name?: string;
   description?: string;
   middleware_address?: string;
+  location_type?: string;
```
Comment (rithviknishad) [inline src/Components/Facility/models.tsx]: Stricter types

```suggestion
  location_type?: "WARD" | "ICU" | "OTHER"
  // or if these choices are defined in an array somewhere already, then:
  location_type?: (typeof LocationTypeChoices)[number]
```

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
@GokulramGHV is it good for testing

**Comment by GokulramGHV** [comment]
@nihal467 There are a few changes to be done, I'll complete and push the changes and then I'll let you know

**Comment by GokulramGHV** [comment]
@nihal467 As per the discussion with @gigincg, we can work on the other changes in the card such as colours for labels and displaying bed count by addressing them in separate issues after discussing with @aparnacoronasafe again.
This PR is ready to be tested.

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**